Helper/ScopeSetting.php

```diff
@@ -194,4 +206,32 @@ public function getOptInSetting()
             return self::API_MEMBERS;
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * This maps a klaviyo account to all the store ids its scoped too.
+     * @param $storeIds
+     * @return array
+     */
+    public function getStoreIdKlaviyoAccountSetMap($storeIds)
+    {
+
+        $storeMap = array();
+        foreach ($storeIds as $storeId) {
+            $klaviyoAccount = $this->getPublicApiKey($storeId);
+            if (!array_key_exists($klaviyoAccount, $storeMap)) {
+                $storeMap[$klaviyoAccount] = array($storeId);
+            } else {
+                array_push( $storeMap[$klaviyoAccount], $storeId);
+            }
+        }
+
+        return $storeMap;
+    }
+
+    public function getProductDeleteBeforeSetting($storeId)
+    {
+        return $this->getScopeSetting(self::PRODUCT_DELETE_BEFORE, $storeId);
+    }
+
+}
+
```

Comment:
nit: i think we have two extra lines here? i believe gh doesn't show the newline at the end of file but i could be wrong